src/Hosting/TestHost/src/RequestLifetimeFeature.cs

```diff
@@ -9,14 +10,25 @@ namespace Microsoft.AspNetCore.TestHost
     internal class RequestLifetimeFeature : IHttpRequestLifetimeFeature
     {
         private readonly CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
+        private readonly Action<Exception> _abort;
 
-        public RequestLifetimeFeature()
+        public RequestLifetimeFeature(Action<Exception> abort)
         {
             RequestAborted = _cancellationTokenSource.Token;
+            _abort = abort;
         }
 
         public CancellationToken RequestAborted { get; set; }
 
-        public void Abort() => _cancellationTokenSource.Cancel();
+        internal void Cancel()
+        {
+            _cancellationTokenSource.Cancel();
```

Comment:
I know it's not exactly new, but I'd be tempted to threadpool dispatch this for the same reason  we use TaskCreationOptions.RunAsync.

Reply:
@halter73 you know better than to fix things that aren't broken 😁

Reply:
That's why I approved it and left the actually choice on whether to do it to you 😉 . I get that kind of change is likely to break existing tests if we're not careful, but it would likely make new tests easier to reason about and debug.